common/app/routes/challenges/components/Classic-Modal.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { PropTypes } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import PureComponent from 'react-pure-render/component';
+import FontAwesome from 'react-fontawesome';
+
+const propTypes = {
+  close: PropTypes.func,
+  keyMap: PropTypes.object,
+  mapKeys: PropTypes.func,
+  open: PropTypes.bool.isRequired,
+  submitChallenge: PropTypes.func.isRequired,
+  successMessage: PropTypes.string.isRequired
+};
+
+export default class ClassicModal extends PureComponent {
+  constructor(...props) {
+    super(...props);
+    this.keyBindings = this.keyBindings.bind(this);
+  }
+  keyBindings(event) {
+    const map = Object.assign({}, this.props.keyMap);
+    const { open, submitChallenge, mapKeys } = this.props;
+    if (event.metaKey) {
+      map['meta'] = event.type === 'keydown';
+    }
+    map[event.keyCode] = event.type === 'keydown';
+    // (Ctrl || Cmd) + Enter
+    if ((map[17] || map['meta']) && map[13] && open) {
+      submitChallenge();
+    }
+    mapKeys(map);
+  }
+
+
+  render() {
+    let {
+      close,
+      open,
+      submitChallenge,
+      successMessage
+    } = this.props;
+    return (
+      <Modal
+        animation={ false }
+        dialogClassName='success'
+        keyboard={ true }
+        onHide={ close }
+        onKeyDown={ this.keyBindings }
+        onKeyUp={ this.keyBindings }
```

Comment:
This needs to be called on keyup also, to set the keyMap prop to false.

This is so that a user cannot press `Ctrl || Cmd` and then `Enter` individually to submit the challenge. The `Ctrl` and `Enter` both need to be keydown to submit the challenge.